### DISCUSSION TITLE
Trying to wrap my ahead around my cluster's storage usage - need some insight

### DISCUSSION THREAD

**Discussion by ragegrammer** [minio/minio#12598] [Q&A] [answered]
Hello,

I've been getting alert notifications due to an issue with high storage usage on our cluster. Checking the reports, it appears that we've hit over 80% total usage. 

Before I get any farther, here is my cluster configuration

minio-01 - 10 X 8TB 
minio-02 - 10 X 8TB

So in total, I have 20 8TB drives running in distributed mode. The TOTAL SIZE is 160TB (not literally 160TB obviously due to OP) but in the current configuration, we're looking at:

10 x 8 TB - TOTAL DATA CAPACITY = 80TB
10 x 8 TB - Parity = 80 TB 

Hopefully I'm still on the right track with my assumptions still about the configuration. It's been awhile since I've configured minio and it's been happily chugging along for the past 2 years now. 

Upon checking the cluster using mc, I get the following readout:

```
mc admin info myminio

●  MINIO-02:80
   Uptime: 6 months
   Version: 2020-07-24T22:43:05Z
   Network: 2/2 OK
   Drives: 10/10 OK

●  MINIO-01:80
   Uptime: 6 months
   Version: 2020-07-24T22:43:05Z
   Network: 2/2 OK
   Drives: 10/10 OK

34 TiB Used, 8 Buckets, 5,959,516 Objects
20 drives online, 0 drives offline
```

This leaves me very confused because my notifications are talking about 80% utilization yet I'm apparently at sub 50%. Checking Grafana, it looks like TOTAL CAPACITY (160TB) is being measured (usage and total space) and I've hit 113TB (according to Grafana).

So I did some digging and calculated the size of each bucket and summed them up. 

```
Total bytes	| Total TiB
6.46892E+13	| 58.83445783
```

This was running the following command across every bucket and combining the results

```
/mc ls -r --json myminio/mybucket | ./jq -s 'map(.size) | add'
```

This is where I'm confused.  What is the 34TiB being reported from mc admin info and why is that different from the 58TiB I came up with when I summed the size of each bucket in my cluster? Is the above command the correct way to get the size of each bucket?

The 58TiB looks to be a lot more close to the 80% utilization of the total capacity (160TiB) given that 58TiB X 2 = 117 TiB (58 TiB in data, 58TiB in parity). 

I'd appreciate it if someone could explain the difference between the size reported by mc admin info versus summing up the size of each bucket in the cluster or am I missing something?

Any insight would be extremely helpful.

**Comment by harshavardhana** [comment, marked as answer]
> This is where I'm confused. What is the 34TiB being reported from mc admin info and why is that different from the 58TiB I came up with when I summed the size of each bucket in my cluster? Is the above command the correct way to get the size of each bucket?

Because we do active scanning of namespace slowly to build the results, it is not instantaneous and it won't show you right away 58TiB - if it's a large enough namespace give it 1-2days. 

> mc admin info myminio
> 
> ●  MINIO-02:80
>    Uptime: 6 months
>    Version: 2020-07-24T22:43:05Z
>    Network: 2/2 OK
>    Drives: 10/10 OK
> 
> ●  MINIO-01:80
>    Uptime: 6 months
>    Version: 2020-07-24T22:43:05Z
>    Network: 2/2 OK
>    Drives: 10/10 OK
> 
> 34 TiB Used, 8 Buckets, 5,959,516 Objects
> 20 drives online, 0 drives offline

Also do not source compile yourself if you do then you are on your own @ragegrammer - please use the released binaries for us to help out.

**Comment by klauspost** [comment]
... and use a newer release. A lot of things have been updated in the last year.
